guest_crates/mycelia_http/Cargo.toml

```diff
@@ -0,0 +1,25 @@
+[package]
+name = "mycelia_http"
+version = "0.1.0"
+edition = "2021"
+license = "MIT OR Apache-2.0"
+description = "guest side wasm http-client for general use in a compatible wasm component host. See docs for more info!"
+authors = ["Hazel Rowell <hazel+mycelia@departure.dev>"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+thiserror = { workspace = true }
+async-trait = { workspace = true }
+anyhow = { workspace = true }
```

Comment:
How do you decide which crate to use for which type of Errors?

I've been reading [this article](https://www.shakacode.com/blog/thiserror-anyhow-or-how-i-handle-errors-in-rust-apps/#:~:text=Use%20thiserror%20%2C%20when%20callers%20are,it%20to%20the%20logging%20pipeline.) and I quote:

> When to use which? That's how I've been making my decisions:

> Use thiserror, when callers are interested in error details. E.g., if an error is an enum, a caller has to handle each error variant differently.
> Use anyhow, when the caller doesn't care about the error details. E.g., it just propagates it to the logging pipeline.
> I've been using both libraries, and my experience with anyhow turned out not to be great. While it is convenient operating on a single type, error handling in apps becomes messy over time, which makes it hard to hold errors quality under control across the app.

Reply:
I've really been enjoying thiserror. I'm not sure I'm confident enough quite yet to make a total switch tho